Adding OctoAI generative and text2vec modules

tmoreau89 opened the pull request:
### What's being changed:
This PR adds OctoAI `generative` and `text2vec` modules.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

tmoreau89 commented:
CC @zainhas

antas-marcin commented on modules/generative-octoai/clients/octoai.go:
```diff
@@ -0,0 +1,222 @@
+		{"role": "user", "content": prompt},
+	}
+
+	// message := Message{
+	// 	Role:    "user",
+	// 	Content: prompt,
+	// }
```
This commented code can be removed

tmoreau89 replied:
Absolutely, thanks for the catch

antas-marcin commented on modules/generative-octoai/clients/octoai.go:
```diff
@@ -0,0 +1,222 @@
+}
+
+func (v *octoai) getValueFromContext(ctx context.Context, key string) string {
+	if value := ctx.Value(key); value != nil {
+		if keyHeader, ok := value.([]string); ok && len(keyHeader) > 0 && len(keyHeader[0]) > 0 {
+			return keyHeader[0]
+		}
+	}
+	// try getting header from GRPC if not successful
+	if apiKey := modulecomponents.GetValueFromGRPC(ctx, key); len(apiKey) > 0 && len(apiKey[0]) > 0 {
+		return apiKey[0]
+	}
```
This part can be replaced with:
```
if apiKeyValue := modulecomponents.GetValueFromContext(ctx, key); apiKeyValue != "" {
		return apiKeyValue
}
```

tmoreau89 replied:
Ack, thank you for the recommendation

antas-marcin commented on modules/generative-octoai/clients/octoai.go:
```diff
@@ -0,0 +1,222 @@
+		return v.apiKey, nil
+	}
+	return "", errors.New("no api key found " +
+		"neither in request header: X-Octoai-Api-Key " +
+		"nor in environment variable under OPENAI_API_KEY")
```
You should change `X-Octoai-Api-Key` to `X-OctoAI-Api-Key` and `OPENAI_API_KEY` to `OCTOAI_APIKEY`

tmoreau89 replied:
Ack on `X-OctoAI-Api-Key`

And whoops that should have been changed from OPENAI -> OCTOAI for sure. That said, we use `OCTOAI_API_KEY`instead of `OCTOAI_APIKEY` a lot in our documentation and code examples (e.g. https://octo.ai/docs/text-gen-solution/using-unstructured-io-for-embedding-documents). I'm happy to use the prescribed formatting for it to work best for Weaviate users but I figured I'd point this out.

antas-marcin commented on modules/generative-octoai/config/class_settings.go:
```diff
@@ -0,0 +1,116 @@
+
+var availableOctoAIModels = []string{
+	"qwen1.5-32b-chat",
+	"meta-llama-3-8b-instruct",
+	"meta-llama-3-70b-instruct",
```
uuuuu, nice!

tmoreau89 replied:
indeed ;)

antas-marcin commented on modules/generative-octoai/module.go:
```diff
@@ -0,0 +1,97 @@
+func (m *GenerativeOctoAIModule) initAdditional(ctx context.Context, timeout time.Duration,
+	logger logrus.FieldLogger,
+) error {
+	apiKey := os.Getenv("OCTOAI_API_KEY")
```
Let's use `OCTOAI_APIKEY` instead of `OCTOAI_API_KEY`

tmoreau89 replied:
ack

antas-marcin commented on modules/text2vec-octoai/module.go:
```diff
@@ -0,0 +1,151 @@
+func (m *OctoAIModule) initVectorizer(ctx context.Context, timeout time.Duration,
+	logger logrus.FieldLogger,
+) error {
+	octoAIApiKey := os.Getenv("OCTOAI_API_KEY")
```
Let's use `OCTOAI_APIKEY` instead of `OCTOAI_API_KEY`

tmoreau89 replied:
ack

antas-marcin commented on test/docker/compose.go:
```diff
@@ -492,6 +511,9 @@ func (d *Compose) Start(ctx context.Context) (*DockerCompose, error) {
 		envSettings["PALM_APIKEY"] = d.withPaLMApiKey
 	}
+	if d.withOctoAIApiKey != "" {
+		envSettings["OCTOAI_API_KEY"] = d.withOctoAIApiKey
```
should be `OCTOAI_APIKEY`

tmoreau89 replied:
ack

antas-marcin commented on test/modules/text2vec-octoai/setup_test.go:
```diff
@@ -0,0 +1,52 @@
+)
+
+func TestText2VecOctoAI_SingleNode(t *testing.T) {
+	octoAIApiKey := os.Getenv("OCTOAI_API_KEY")
```
should be `OCTOAI_APIKEY`

tmoreau89 commented on modules/text2vec-octoai/clients/octoai.go:
```diff
@@ -0,0 +1,252 @@
+	var apiKey, envVar string
+
+	apiKey = "X-Octoai-Api-Key"
+	envVar = "OCTOAI_API_KEY"
```
Ack

tmoreau89 commented on test/modules/generative-octoai/setup_test.go:
```diff
@@ -0,0 +1,52 @@
+)
+
+func TestGenerativeOctoAI_SingleNode(t *testing.T) {
+	octoAIApiKey := os.Getenv("OCTOAI_API_KEY")
```
ack

tmoreau89 commented:
Thanks for the review @antas-marcin !

At the moment Quality Gate fails because of code duplication that is deemed too high. This is mainly because in order to support the `generative-octoai` and `text2vec-octoai` modules I borrowed heavily from existing implementations (anyscale, mistral, jinaai, ollama, palm etc.). Do you have recommendations there on how to meet the quality requirements?

Finally, what's the best way to test on a simple weaviate example (e.g. https://weaviate.io/developers/weaviate/quickstart) that we can with these changes make use of the OctoAI `generative` and `text2vec` modules? Thanks for your advice.

tmoreau89 commented:
@antas-marcin @zainhas quick end of day updates. I was able to make sure that the following tests are working as intended:
```
OCTOAI_APIKEY=$OCTOAI_API_TOKEN go test -count 1 -v ./test/modules/text2vec-octoai
OCTOAI_APIKEY=$OCTOAI_API_TOKEN go test -count 1 -v ./test/modules/generative-octoai
```

Which produces the following results
```
=== RUN   TestText2VecOctoAI_SingleNode/tests
=== RUN   TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large
=== RUN   TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/create_objects
=== RUN   TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/check_objects_existence
=== RUN   TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/check_objects_existence/00000000-0000-0000-0000-000000000001
=== RUN   TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/check_objects_existence/00000000-0000-0000-0000-000000000002
=== RUN   TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/perform_vector_search
2024/04/30 18:32:56 🐳 Terminating container: dd1c46b938e5
2024/04/30 18:32:56 🚫 Container terminated: dd1c46b938e5
--- PASS: TestText2VecOctoAI_SingleNode (7.93s)
    --- PASS: TestText2VecOctoAI_SingleNode/tests (0.32s)
        --- PASS: TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large (0.32s)
            --- PASS: TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/create_objects (0.23s)
            --- PASS: TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/check_objects_existence (0.01s)
                --- PASS: TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/check_objects_existence/00000000-0000-0000-0000-000000000001 (0.00s)
                --- PASS: TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/check_objects_existence/00000000-0000-0000-0000-000000000002 (0.00s)
            --- PASS: TestText2VecOctoAI_SingleNode/tests/thenlper/gte-large/perform_vector_search (0.06s)
PASS
ok      github.com/weaviate/weaviate/test/modules/text2vec-octoai       8.187s
```
and 
```
=== RUN   TestGenerativeOctoAI_SingleNode/tests
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/create_objects
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/check_objects_existence
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/create_a_tweet
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/create_objects
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/check_objects_existence
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002
=== RUN   TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/create_a_tweet
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/create_objects
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/check_objects_existence
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/create_a_tweet
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/create_objects
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/check_objects_existence
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/create_a_tweet
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/create_objects
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/check_objects_existence
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002
=== RUN   TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/create_a_tweet
=== RUN   TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct
=== RUN   TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/create_objects
=== RUN   TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/check_objects_existence
=== RUN   TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001
=== RUN   TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002
=== RUN   TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/create_a_tweet
2024/04/30 18:31:52 🐳 Terminating container: 3e933cf4cb4e
2024/04/30 18:31:52 🚫 Container terminated: 3e933cf4cb4e
--- PASS: TestGenerativeOctoAI_SingleNode (43.13s)
    --- PASS: TestGenerativeOctoAI_SingleNode/tests (12.05s)
        --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct (3.53s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/create_objects (1.45s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/check_objects_existence (0.00s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001 (0.00s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002 (0.00s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-8b-instruct/create_a_tweet (2.03s)
        --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct (1.77s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/create_objects (0.14s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/check_objects_existence (0.01s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001 (0.00s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002 (0.00s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/meta-llama-3-70b-instruct/create_a_tweet (1.60s)
        --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct (1.63s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/create_objects (0.11s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/check_objects_existence (0.00s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001 (0.00s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002 (0.00s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x22b-instruct/create_a_tweet (1.47s)
        --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct (1.70s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/create_objects (0.13s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/check_objects_existence (0.01s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001 (0.01s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002 (0.00s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/mixtral-8x7b-instruct/create_a_tweet (1.52s)
        --- PASS: TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct (1.70s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/create_objects (0.13s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/check_objects_existence (0.00s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001 (0.00s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002 (0.00s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/mistral-7b-instruct/create_a_tweet (1.52s)
        --- PASS: TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct (1.71s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/create_objects (0.13s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/check_objects_existence (0.01s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000001 (0.01s)
                --- PASS: TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/check_objects_existence/00000000-0000-0000-0000-000000000002 (0.01s)
            --- PASS: TestGenerativeOctoAI_SingleNode/tests/codellama-34b-instruct/create_a_tweet (1.53s)
PASS
ok      github.com/weaviate/weaviate/test/modules/generative-octoai     43.457s
```

Are there other tests you recommend running?

tmoreau89 commented:
Fixed linter errors, and added `X-OctoAI-Api-Key` header in `config_handler.go`

Also ran the following tests for each module:

```
OCTOAI_APIKEY=$OCTOAI_API_TOKEN go test -count 1 -v ./modules/generative-octoai
```

```
=== RUN   TestGetMeta
=== RUN   TestGetMeta/when_the_server_is_providing_meta
--- PASS: TestGetMeta (0.00s)
    --- PASS: TestGetMeta/when_the_server_is_providing_meta (0.00s)
=== RUN   TestGetAnswer
=== RUN   TestGetAnswer/when_the_server_has_a_successful_answer
=== RUN   TestGetAnswer/when_the_server_has_an_error
=== RUN   TestGetAnswer/when_the_server_does_not_respond_in_time
=== RUN   TestGetAnswer/when_X-Octoai-BaseURL_header_is_passed
--- PASS: TestGetAnswer (1.01s)
    --- PASS: TestGetAnswer/when_the_server_has_a_successful_answer (0.00s)
    --- PASS: TestGetAnswer/when_the_server_has_an_error (0.00s)
    --- PASS: TestGetAnswer/when_the_server_does_not_respond_in_time (1.00s)
    --- PASS: TestGetAnswer/when_X-Octoai-BaseURL_header_is_passed (0.00s)
PASS
ok      github.com/weaviate/weaviate/modules/generative-octoai/clients  1.344s
```

```
OCTOAI_APIKEY=$OCTOAI_API_TOKEN go test -count 1 -v ./modules/generative-octoai/config
```

```
=== RUN   Test_classSettings_Validate
=== RUN   Test_classSettings_Validate/default_settings
=== RUN   Test_classSettings_Validate/everything_non_default_configured
=== RUN   Test_classSettings_Validate/wrong_model_configured
=== RUN   Test_classSettings_Validate/default_settings_with_meta-llama-3-8b-instruct_and_baseURL
--- PASS: Test_classSettings_Validate (0.00s)
    --- PASS: Test_classSettings_Validate/default_settings (0.00s)
    --- PASS: Test_classSettings_Validate/everything_non_default_configured (0.00s)
    --- PASS: Test_classSettings_Validate/wrong_model_configured (0.00s)
    --- PASS: Test_classSettings_Validate/default_settings_with_meta-llama-3-8b-instruct_and_baseURL (0.00s)
PASS
ok      github.com/weaviate/weaviate/modules/generative-octoai/config   0.118s
```

```
$OCTOAI_API_TOKEN go test -count 1 -v ./modules/text2vec-octoai/clients
```

```
=== RUN   TestBuildUrlFn
=== RUN   TestBuildUrlFn/buildUrlFn_returns_default_OctoAI_URL
=== RUN   TestBuildUrlFn/buildUrlFn_loads_from_BaseURL
--- PASS: TestBuildUrlFn (0.00s)
    --- PASS: TestBuildUrlFn/buildUrlFn_returns_default_OctoAI_URL (0.00s)
    --- PASS: TestBuildUrlFn/buildUrlFn_loads_from_BaseURL (0.00s)
=== RUN   TestClient
=== RUN   TestClient/when_all_is_fine
=== RUN   TestClient/when_the_context_is_expired
=== RUN   TestClient/when_the_server_returns_an_error
=== RUN   TestClient/when_OctoAI_key_is_passed_using_X-OctoAI-Api-Key_header
=== RUN   TestClient/when_OctoAI_key_is_empty
=== RUN   TestClient/when_X-OctoAI-Api-Key_header_is_passed_but_empty
--- PASS: TestClient (0.00s)
    --- PASS: TestClient/when_all_is_fine (0.00s)
    --- PASS: TestClient/when_the_context_is_expired (0.00s)
    --- PASS: TestClient/when_the_server_returns_an_error (0.00s)
    --- PASS: TestClient/when_OctoAI_key_is_passed_using_X-OctoAI-Api-Key_header (0.00s)
    --- PASS: TestClient/when_OctoAI_key_is_empty (0.00s)
    --- PASS: TestClient/when_X-OctoAI-Api-Key_header_is_passed_but_empty (0.00s)
PASS
ok      github.com/weaviate/weaviate/modules/text2vec-octoai/clients    0.173s
```

```
OCTOAI_APIKEY=$OCTOAI_API_TOKEN go test -count 1 -v ./modules/text2vec-octoai/vectorizer
```

```
=== RUN   TestVectorizingObjects
=== RUN   TestVectorizingObjects/empty_object
=== RUN   TestVectorizingObjects/object_with_one_string_prop
=== RUN   TestVectorizingObjects/object_with_one_non-string_prop
=== RUN   TestVectorizingObjects/object_with_a_mix_of_props
=== RUN   TestVectorizingObjects/with_a_noindexed_property
=== RUN   TestVectorizingObjects/with_the_class_name_not_vectorized
=== RUN   TestVectorizingObjects/with_a_property_name_not_vectorized
=== RUN   TestVectorizingObjects/with_no_schema_labels_vectorized
=== RUN   TestVectorizingObjects/with_string/text_arrays_without_propname_or_classname
=== RUN   TestVectorizingObjects/with_string/text_arrays_with_propname_and_classname
=== RUN   TestVectorizingObjects/with_compound_class_and_prop_names
--- PASS: TestVectorizingObjects (0.00s)
    --- PASS: TestVectorizingObjects/empty_object (0.00s)
    --- PASS: TestVectorizingObjects/object_with_one_string_prop (0.00s)
    --- PASS: TestVectorizingObjects/object_with_one_non-string_prop (0.00s)
    --- PASS: TestVectorizingObjects/object_with_a_mix_of_props (0.00s)
    --- PASS: TestVectorizingObjects/with_a_noindexed_property (0.00s)
    --- PASS: TestVectorizingObjects/with_the_class_name_not_vectorized (0.00s)
    --- PASS: TestVectorizingObjects/with_a_property_name_not_vectorized (0.00s)
    --- PASS: TestVectorizingObjects/with_no_schema_labels_vectorized (0.00s)
    --- PASS: TestVectorizingObjects/with_string/text_arrays_without_propname_or_classname (0.00s)
    --- PASS: TestVectorizingObjects/with_string/text_arrays_with_propname_and_classname (0.00s)
    --- PASS: TestVectorizingObjects/with_compound_class_and_prop_names (0.00s)
=== RUN   TestClassSettings
--- PASS: TestClassSettings (0.00s)
=== RUN   TestVectorizingTexts
=== RUN   TestVectorizingTexts/single_word
=== RUN   TestVectorizingTexts/multiple_words
=== RUN   TestVectorizingTexts/multiple_sentences_(joined_with_a_dot)
=== RUN   TestVectorizingTexts/multiple_sentences_already_containing_a_dot
=== RUN   TestVectorizingTexts/multiple_sentences_already_containing_a_question_mark
=== RUN   TestVectorizingTexts/multiple_sentences_already_containing_an_exclamation_mark
=== RUN   TestVectorizingTexts/multiple_sentences_already_containing_comma
--- PASS: TestVectorizingTexts (0.00s)
    --- PASS: TestVectorizingTexts/single_word (0.00s)
    --- PASS: TestVectorizingTexts/multiple_words (0.00s)
    --- PASS: TestVectorizingTexts/multiple_sentences_(joined_with_a_dot) (0.00s)
    --- PASS: TestVectorizingTexts/multiple_sentences_already_containing_a_dot (0.00s)
    --- PASS: TestVectorizingTexts/multiple_sentences_already_containing_a_question_mark (0.00s)
    --- PASS: TestVectorizingTexts/multiple_sentences_already_containing_an_exclamation_mark (0.00s)
    --- PASS: TestVectorizingTexts/multiple_sentences_already_containing_comma (0.00s)
PASS
ok      github.com/weaviate/weaviate/modules/text2vec-octoai/vectorizer 0.178s
```

antas-marcin commented on usecases/config/config_handler.go:
```diff
@@ -273,7 +273,7 @@ type CORS struct {
 	DefaultCORSAllowOrigin  = "*"
 	DefaultCORSAllowMethods = "*"
-	DefaultCORSAllowHeaders = "Content-Type, Authorization, Batch, X-Openai-Api-Key, X-Openai-Organization, X-Openai-Baseurl, X-Anyscale-Baseurl, X-Anyscale-Api-Key, X-Cohere-Api-Key, X-Cohere-Baseurl, X-Huggingface-Api-Key, X-Azure-Api-Key, X-Google-Api-Key, X-Palm-Api-Key, X-Jinaai-Api-Key, X-Aws-Access-Key, X-Aws-Secret-Key, X-Voyageai-Baseurl, X-Voyageai-Api-Key, X-Mistral-Baseurl, X-Mistral-Api ...
+	DefaultCORSAllowHeaders = "Content-Type, Authorization, Batch, X-Openai-Api-Key, X-Openai-Organization, X-Openai-Baseurl, X-Anyscale-Baseurl, X-Anyscale-Api-Key, X-Cohere-Api-Key, X-Cohere-Baseurl, X-Huggingface-Api-Key, X-Azure-Api-Key, X-Google-Api-Key, X-Palm-Api-Key, X-Jinaai-Api-Key, X-Aws-Access-Key, X-Aws-Secret-Key, X-Voyageai-Baseurl, X-Voyageai-Api-Key, X-Mistral-Baseurl, X-Mistral-Api ...
```
you need to also add `X-Octoai-Baseurl` here
